From 21d193506851e0727860927ab289869732b06102 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Wed, 21 Jun 2023 12:39:15 +0200
Subject: [PATCH] input: don't crash if the internal handle is nullptr

Pointer focus can stay on a closed tooltip while focus updates are blocked,
so this needs to be checked for

BUG: 471285


(cherry picked from commit c25aaa2c9fbf8ec10f1ba16fecd4b31704fdaf0c)
---
 src/input.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/input.cpp b/src/input.cpp
index b4940f123ad..d3d32020580 100644
--- a/src/input.cpp
+++ b/src/input.cpp
@@ -1185,6 +1185,10 @@ class InternalWindowEventFilter : public InputEventFilter
             return false;
         }
         QWindow *internal = static_cast<InternalWindow *>(input()->pointer()->focus())->handle();
+        if (!internal) {
+            // the handle can be nullptr if the tooltip gets closed while focus updates are blocked
+            return false;
+        }
         QMouseEvent mouseEvent(event->type(),
                                event->pos() - internal->position(),
                                event->globalPos(),
-- 
GitLab

